src/Utility/Text.php

```diff
@@ -891,7 +891,9 @@ public static function excerpt(string $text, string $phrase, int $radius = 100,
      */
     public static function toList(array $list, ?string $and = null, string $separator = ', '): string
     {
-        $and ??= __d('cake', 'and');
+        $defaultAnd = function_exists('__d') ? __d('cake', 'and') : 'and';
```

Comment:
```suggestion
        static $defaultAnd ??= function_exists('__d') ? __d('cake', 'and') : 'and';
```

This should avoid the function_exits() call on each invocation of the toList() method.

Reply:
A static variable would solve the problem, but before PHP 8.3, you couldn't assign values from dynamic expressions to static variables (see [doc](https://www.php.net/manual/en/language.variables.scope.php)). 
The requirements for version 5.x are [>= 8.1](https://github.com/cakephp/cakephp/blob/5.x/composer.json#L25)

Reply:
Guess we need to use a static property then.